src/javascript/esutil.ts

```diff
@@ -199,6 +199,109 @@ export function getPropertyValue(
   }
 }
 
+export function getDescription(jsdocAnn: jsdoc.Annotation): string|undefined {
+  if (jsdocAnn.description) {
+    return jsdocAnn.description;
+  }
+  // These tags can be used to describe a field.
+  // e.g.:
+  //    /** @type {string} the name of the animal */
+  //    this.name = name || 'Rex';
+  const tagSet = new Set(['public', 'private', 'protected', 'type']);
+  for (const tag of jsdocAnn.tags) {
+    if (tagSet.has(tag.title) && tag.description) {
+      return tag.description;
+    }
+  }
+}
+
+export function propertiesFromGettersAndSetters(
+    body: Array<babel.Method|babel.Property>, document: JavaScriptDocument): ScannedProperty[] {
+  const properties: ScannedProperty[] = [];
+  const accessors = new Map<
+      string,
+      {getter?: babel.Method, setter?: babel.Method}>();
+
+  for (const member of body) {
+    if (!babel.isMethod(member)) {
+      continue;
+    }
+
+    // TODO: we dont support statics yet
+    if (babel.isClassMethod(member) && member.static) {
+      continue;
+    }
+
+    if (!babel.isIdentifier(member.key) || !member.key.name) {
```

Comment:
The member name shouldn't be computed either. e.g. a method defined like this would match I think unless we check for `computed`, and we'd think that the method name was `dynamicName` rather than `foo`:

```javascript
const dynamicName = 'foo';
const obj = {
  [dynamicName]() {
    return 'bar';
  }
};
```

https://astexplorer.net/#/gist/3b8e43eae071c9aab4cbdf243b8a744e/fb585ceff75671e1ffd27e23bf65c4f969ff0acb